Redo Fabric Loom update back to 1.5, Fix compiling errors

Kichura opened the pull request:
Title says it, Might aswell help out instead of giving up at trying.

Kichura committed:
Update Fabric Loom to 1.5.

Kichura committed:
Update MixinExtras to 0.3.5.

DragonEggBedrockBreaking reviewed (changes requested):
On fabric I am using the mixin-extras version bundled in fabric loader (0.3.3), please could you set the mixin-extras version on neoforge to the same

Kichura committed:
Downgrade MixinExtras to 0.3.3.

As requested by DragonEggBedrockBreaking, However 0.3.3 is normally not recommended as this version caused problems in fabric loader 0.15.4 for certain mods.
